src/Messages/OWSDisappearingMessagesJob.m

```diff
@@ -121,7 +121,7 @@ - (void)runLoop
 
 + (void)setExpirationForMessage:(TSMessage *)message
 {
-    dispatch_async(self.serialQueue, ^{
+    dispatch_sync(self.serialQueue, ^{
```

Comment:
Yup, that's why it's not merged. 

I think I'd prefer to add a completion handler vs. getting in the habit of adding sleep(1) in tests, but that would take more than a minute of dev time. 

This PR hasn't been top priority for me, so feel free to run with it if you're feeling frisky.